spec/models/student_spec.rb

```diff
@@ -1,5 +1,19 @@
 require 'rails_helper'
 
-RSpec.describe Student, :type => :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+RSpec.describe Student, type: :model do
+  
+  context 'with valid attributes' do
+    let(:student) { build(:student) }
+    
+    subject { student }
+    it "is valid" do
+      expect(student).to be_valid
+    end
+    
+    subject { student }
+    it "returns full name as a string" do
+      expect(subject.full_name).to eq("#{subject.title} #{subject.first_name} #{subject.middle_name} #{subject.last_name}")
+    end
+
+  end
```

Comment:
There's no need to redefined subject here, and we can move it above the `context` blocks.

Also forgot to pick up on this originally, but we prefer to use literal values when making assertions (see the specs section on [this confluence page](https://jobready.atlassian.net/wiki/spaces/DEV/pages/43941936/Code+Review+Checklist)).

```suggestion
RSpec.describe Student, type: :model do
  subject { student }
  
  context 'with valid attributes' do
    let(:student) { build(:student, title: "Mr", first: "first", middle: "middle", last: "last") }
    
    it "is valid" do
      expect(student).to be_valid
    end
    
    it "returns full name as a string" do
      expect(subject.full_name).to eq("Mr first middle last")
    end

  end
```

the block is lazily loaded so it doesn't matter that the subject is defined before `student` is

Reply:
Thank - didn't realise about the literal values. Will cross-check with checklist from now on!